src/mumble/MainWindow.cpp

```diff
@@ -63,37 +63,26 @@ OpenURLEvent::OpenURLEvent(QUrl u) : QEvent(static_cast<QEvent::Type>(OU_QEVENT)
 }
 
 MainWindow::MainWindow(QWidget *p) : QMainWindow(p) {
-	qiIconMuteSelf.addFile(QLatin1String("skin:muted_self.svg"));
-	qiIconMuteServer.addFile(QLatin1String("skin:muted_server.svg"));
-	qiIconMuteSuppressed.addFile(QLatin1String("skin:muted_suppressed.svg"));
-	qiIconMutePushToMute.addFile(QLatin1String("skin:muted_pushtomute.svg"));
-	qiIconDeafSelf.addFile(QLatin1String("skin:deafened_self.svg"));
-	qiIconDeafServer.addFile(QLatin1String("skin:deafened_server.svg"));
-	qiTalkingOff.addFile(QLatin1String("skin:talking_off.svg"));
-	qiTalkingOn.addFile(QLatin1String("skin:talking_on.svg"));
-	qiTalkingShout.addFile(QLatin1String("skin:talking_alt.svg"));
-	qiTalkingWhisper.addFile(QLatin1String("skin:talking_whisper.svg"));
+	qiIconMuteSelf.load(QLatin1String("skin:muted_self.svg"));
+	qiIconMuteServer.load(QLatin1String("skin:muted_server.svg"));
+	qiIconMuteSuppressed.load(QLatin1String("skin:muted_suppressed.svg"));
+	qiIconMutePushToMute.load(QLatin1String("skin:muted_pushtomute.svg"));
+	qiIconDeafSelf.load(QLatin1String("skin:deafened_self.svg"));
+	qiIconDeafServer.load(QLatin1String("skin:deafened_server.svg"));
+	qiTalkingOff.load(QLatin1String("skin:talking_off.svg"));
+	qiTalkingOn.load(QLatin1String("skin:talking_on.svg"));
+	qiTalkingShout.load(QLatin1String("skin:talking_alt.svg"));
+	qiTalkingWhisper.load(QLatin1String("skin:talking_whisper.svg"));
 
 #ifdef Q_OS_MAC
 	if (QFile::exists(QLatin1String("skin:mumble.icns")))
-		qiIcon.addFile(QLatin1String("skin:mumble.icns"));
+		qiIcon.load(QLatin1String("skin:mumble.icns"));
 	else
-		qiIcon.addFile(QLatin1String("skin:mumble.svg"));
+		qiIcon.load(QLatin1String("skin:mumble.svg"));
 #else
 	{
-		QSvgRenderer svg(QLatin1String("skin:mumble.svg"));
-		QPixmap original(512,512);
-		original.fill(Qt::transparent);
-
-		QPainter painter(&original);
-		painter.setRenderHint(QPainter::Antialiasing);
-		painter.setRenderHint(QPainter::TextAntialiasing);
-		painter.setRenderHint(QPainter::SmoothPixmapTransform);
-		painter.setRenderHint(QPainter::HighQualityAntialiasing);
-		svg.render(&painter);
-
-		for (int sz=8;sz<=256;sz+=8)
-			qiIcon.addPixmap(original.scaled(sz,sz, Qt::IgnoreAspectRatio, Qt::SmoothTransformation));
+
+		qiIcon.load(QLatin1String("skin:mumble.svg"));
```

Comment:
It doesn't seem to me that the code is equivalent to the code above.
The old code ensures we have high quality scaling across all icon sizes. The new code uses a pixmap of an unknown size (probably based on the baseline size specified in the SVG? who knows?)

Reply:
I tested the QPixmap code with and without QSvgRenderer, the results appear to be the same.

Maybe it's worth creating an application to test various cases.